lib/tx_build/preconditions.ex

```diff
@@ -0,0 +1,185 @@
+defmodule Stellar.TxBuild.Preconditions do
+  @moduledoc """
+  `Preconditions` struct definition.
+  """
+  alias Stellar.TxBuild.{
+    TimeBounds,
+    LedgerBounds,
+    SequenceNumber,
+    SignerKey
+  }
+
+  alias StellarBase.XDR.TimeBounds, as: TimeBoundsXDR
+
+  alias StellarBase.XDR.{
+    Preconditions,
+    PreconditionsV2,
+    PreconditionType,
+    Duration,
+    OptionalSequenceNumber,
+    SignerKeyList,
+    TimePoint,
+    UInt32,
+    Void
+  }
+
+  @behaviour Stellar.TxBuild.XDR
+
+  @type precond_v2 :: %{
+          time_bounds: TimeBounds.t(),
+          ledger_bounds: LedgerBounds.t(),
+          min_seq_num: SequenceNumber.t(),
+          min_seq_age: non_neg_integer(),
+          min_seq_ledger_gap: non_neg_integer(),
+          extra_signers: list(SignerKey.t())
+        }
+  @type validation ::
```

Comment:
@CristhianRodriguezMolina @FelipeGuzmanSierra Could you please make this line readable? A tip: define a new specification and then just set `validation :: {:ok, valid()} | {:error, atom()}`